saleor/dashboard/order/forms.py

```diff
@@ -181,8 +220,58 @@ def save(self):
             order.change_status('shipped')
 
 
-ORDER_STATUS_CHOICES = (('', pgettext_lazy('Order status field value',
-                                           'All')),) + Order.STATUS_CHOICES
+class CancelGroupForm(forms.Form):
+    def __init__(self, *args, **kwargs):
+        self.delivery_group = kwargs.pop('delivery_group')
+        super(CancelGroupForm, self).__init__(*args, **kwargs)
+
+    def cancel_group(self):
+        for line in self.delivery_group:
+            Stock.objects.deallocate_stock(line.stock, line.quantity)
+        self.delivery_group.status = Status.CANCELLED
+        self.delivery_group.save()
+        other_groups = self.delivery_group.order.groups.all()
+        statuses = other_groups.values_list('status', flat=True)
+        if all(status == Status.CANCELLED for status in statuses):
```

Comment:
Could also be `if set(statuses) == {Status.CANCELLED}:`, not sure which way is more readable.